From 3bcb67720d3523f942d61b5617abb8ad6cc712ba Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 25 Sep 2020 07:55:14 +0300
Subject: [PATCH] [SW-21029] goya2: ppboot: bug fix for phy fw copy

This patch fixes the phy fw copy code where the
variable size is incorrectly used in bytes.
Since the copy happens in u32, size needs to
be accounted in u32 (and not in bytes).

Change-Id: I5ef88f8c3898ad9b5fb8788d1a4b6d6367caf0d9
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/goya2_arc/asic.c b/goya2_arc/asic.c
index 00fb888..4c7d680 100644
--- a/goya2_arc/asic.c
+++ b/goya2_arc/asic.c
@@ -71,7 +71,7 @@
 	if (rst_src != HL_COLD_RST)
 		return;
 
-	for (i = 0; i < size; i++)
+	for (i = 0; i < (size / 4); i++)
 		*(dst + i) = *(src + i);
 }
 
